src/developer-documentation/ibc/ibc.md

```diff
@@ -0,0 +1,999 @@
+        pub account: String,
+        pub channel_id: String,
+    }
+
+    pub enum SpecialQuery {
+        Ping {},
+        Capitalized { text: String },
+    }
+
+    pub enum CustomMsg {
+        Debug(String),
+        Raw(Vec<u8>),
+    }
+
+    pub struct OwnerResponse {
+        pub owner: String,
+    }
+
+    pub struct MessageInfo {
+        /// The `sender` field from `MsgInstantiateContract` and `MsgExecuteContract`.
+        /// You can think of this as the address that initiated the action (i.e. the message). What that
+        /// means exactly heavily depends on the application.
+        ///
+        /// The x/wasm module ensures that the sender address signed the transaction or
+        /// is otherwise authorized to send the message.
+        ///
+        /// Additional signers of the transaction that are either needed for other messages or contain unnecessary
+        /// signatures are not propagated into the contract.
+        pub sender: Addr,
+        /// The funds that are sent to the contract as part of `MsgInstantiateContract`
+        /// or `MsgExecuteContract`. The transfer is processed in bank before the contract
+        /// is executed such that the new balance is visible during contract execution.
+        pub funds: Vec<Coin>,
+    }
+
+    pub struct SubMsgResponse {
+        pub events: Vec<Event>,
+        pub data: Option<Vec<u8>>,
+    }
+
+    pub struct Event {
+        /// The event type. This is renamed to "ty" because "type" is reserved in Rust. This sucks, we know.
+        pub ty: String,
+        /// The attributes to be included in the event.
+        ///
+        /// You can learn more about these from [*Cosmos SDK* docs].
+        ///
+        /// [*Cosmos SDK* docs]: https://docs.cosmos.network/main/core/events.html
+        pub attributes: Vec<Attribute>,
+    }
+
+    pub struct Attribute {
+        pub key: String,
+        pub value: String,
+    }
+
+    pub struct CapitalizedResponse {
+        pub text: String,
+    }
+
+    pub struct ChainResponse {
+        pub data: Vec<u8>,
+    }
+
+    pub struct Coin {
+        pub denom: String,
+        pub amount: u128,
+    }
+
+    pub struct RawResponse {
+        /// The returned value of the raw query. Empty data can be the
+        /// result of a non-existent key or an empty value. We cannot
+        /// differentiate those two cases in cross contract queries.
+        pub data: Vec<u8>,
+    }
+
+    pub enum CosmosMsg<T> {
+        BankMsg(BankMsg),
+        // by default we use RawMsg, but a contract can override that
+        // to call into more app-specific code (whatever they define)
+        Custom(T),
+        // #[cfg(feature = "staking")]
+        Staking(StakingMsg),
+        // #[cfg(feature = "staking")]
+        Distribution(DistributionMsg),
+        // /// A Stargate message encoded the same way as a protobuf [Any](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/any.proto).
```

Comment:
Is there a reason to have `// ///` here for commenting?

Reply:
no, i will remove later